front-end/app/basket/basket.component.html

```diff
@@ -20,7 +20,7 @@
                             <button class="btn btn-warning decrease" [disabled]="item.quantity < 2" (click)='basket.decrease(item)'><span class="glyphicon glyphicon-minus"></span></button>
                         </td>
                         <td>
-                            <button class="btn btn-danger" (click)="basket.removeItem(item)"><span class="glyphicon glyphicon-remove"></span></button>
+                            <button class="btn btn-danger" (click)="removeItems(item)"><span class="glyphicon glyphicon-remove"></span></button>
```

Comment:
It would be nicer if you had one method called removeItem that took an item, and another called clearBasket... Then you also wouldn't need to do `if (item) { ... } else { ... }`